packages/flutter/test/material/slider_theme_test.dart

```diff
@@ -1862,6 +1864,86 @@ void main() {
     expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.text);
   });
 
+  testWidgets('The allowed interaction is themeable', (WidgetTester tester) async {
+    double value = 0.0;
+
+    Widget buildApp({
+      bool isAllowedInteractionInThemeNull = false,
+      bool isAllowedInteractionInSliderNull = false,
+    }) => MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: SliderTheme(
+              data: ThemeData().sliderTheme.copyWith(
+                  allowedInteraction: isAllowedInteractionInThemeNull
+                      ? null
+                      : SliderInteraction.slideOnly,
+              ),
+              child: StatefulBuilder(
+                  builder: (_, void Function(void Function()) setState) {
+                    return Slider(
+                      value: value,
+                      allowedInteraction: isAllowedInteractionInSliderNull
+                          ? null
+                          : SliderInteraction.tapOnly,
+                      onChanged: (double newValue) {
+                        setState(() {
+                          value = newValue;
+                        });
+                      },
+                    );
+                  }
+              ),
+            ),
+          ),
+        ),
+      );
+
+    Future<TestGesture> startCenterTap() => tester.startGesture(tester.getCenter(find.byType(Slider)));
+
+    Future<void> expectTap(bool expectYesOrNo, {required String whileTesting}) async {
+      value = 0.0;
+      final TestGesture gesture = await startCenterTap();
+      await tester.pump();
+      await gesture.cancel();
+      expect(value, expectYesOrNo ? equals(0.5) : isNot(equals(0.5)),
+          reason: 'allowedInteraction is $whileTesting');
+    }
+
+    Future<void> expectSlide(bool expectYesOrNo, {required String whileTesting}) async {
```

Comment:
```suggestion
    Future<void> expectSlide(bool expectYesOrNo, {required String whileTesting}) async {
```
`expectYesOrNo` seems like an unusual wording. There is no such usage in the framework.